src/PubSubClient.cpp

```diff
@@ -537,20 +538,20 @@ uint8_t PubSubClient::buildHeader(uint8_t header, uint8_t* buf, size_t length) {
     } while (len > 0 && hdrLen < MQTT_MAX_HEADER_SIZE - 1);
 
     if (len > 0) {
-        DEBUG_PSC_PRINTF("length too big %zu, left %zu, should be 0\r\n", length, len);
+        ERROR_PSC_PRINTF_P("buildHeader() length too big %zu, left %zu\n", length, len);
+        hdrLen = 0;
```

Comment:
You should return 0 here instead of setting some fake header length.
At the end you return `hdrLen + 1`, which makes a non-zero return value and thus plausible to be non-error.
Just return 0 when it is clear you ran into an error.
This also allows for simply checking like this:
```c++
if (buildHeader(...))
```

Reply:
Wow! Good finding! The `return hdrLen + 1;` really tricked me. ;-)